test/util/job.ts

```diff
@@ -28,15 +39,26 @@ describe('Canceling a job', async function () {
 
   describe('when workflow termination is requested', async function () {
     it('terminates the workflow', async function () {
-      await cancelAndSaveJob(aJob.requestId, 'Canceled by admin', log, true, 'joe');
+      await cancelAndSaveJob(anArgoJob.requestId, 'Canceled by admin', log, true, 'joe');
       expect(terminateWorkflowsStub.callCount).to.equal(1);
     });
+
+    it('does not terminates the workflow', async function () {
+      await cancelAndSaveJob(anotherArgoJob.requestId, 'Canceled by admin', log, false, 'joe');
+      expect(terminateWorkflowsStub.callCount).to.equal(0);
+    });
   });
 
-  describe('when workflow termination is not requested', async function () {
+  describe('when cancelation is requested for a turbo workflow', async function () {
```

Comment:
Add one more test with work items in each of the states and verify it only updates the status to canceled for the ready and running ones.